tests/conftest.py

```diff
@@ -65,33 +64,19 @@ def netcdf_local_paths(daily_xarray_dataset, tmpdir_factory, request):
 
 
 @pytest.fixture()
-def netcdf_http_server(netcdf_local_paths, request):
+def netcdf_http_server(netcdf_local_paths, request, httpserver: HTTPServer):
     def make_netcdf_http_server(username="", password=""):
         first_path = netcdf_local_paths[0]
         # assume that all files are in the same directory
         basedir = first_path.dirpath()
         fnames = [path.basename for path in netcdf_local_paths]
 
-        this_dir = os.path.dirname(os.path.abspath(__file__))
-        port = get_open_port()
-        command_list = [
-            "python",
-            os.path.join(this_dir, "http_auth_server.py"),
-            port,
-            "127.0.0.1",
-            username,
-            password,
-        ]
-        if username:
-            command_list += [username, password]
-        p = subprocess.Popen(command_list, cwd=basedir)
-        url = f"http://127.0.0.1:{port}"
-        time.sleep(1)  # let the server start up
-
-        def teardown():
-            p.kill()
-
-        request.addfinalizer(teardown)
+        url = f"http://127.0.0.1:{httpserver.port}"
+        for fname in fnames:
+            with open(os.path.join(basedir, fname), "rb") as f:
+                data = f.read()
+            httpserver.expect_request(f"/{fname}").respond_with_data(data)
```

Comment:
This loads the file contents in memory, but I couldn't think of another way to do it. Any suggestions?

Reply:
pytest-httpserver allows to register a callable which can return a Response object, so you don't have to load all the files to the memory.

https://pytest-httpserver.readthedocs.io/en/latest/tutorial.html#specifying-responses

```python
def my_handler(request):
    # here, examine the request object
    return Response("Hello world!")

respond_with_handler(my_handler)
```

It uses the request/response objects from werkzeug, for the API docs, see here: https://werkzeug.palletsprojects.com/en/1.0.x/wrappers/

For the authentication: it supports to handle headers, such as the `Authorization` header, the documentation includes example for digest authentication but based on the code, it should work with basic auth as well, you would only need to include the base64 string to the header matcher.

Reply:
Thanks for the help @csernazs! I'll update this PR with your suggestions.